### PR TITLE
Sort upcoming events by starttime on the homepage

### PR DESCRIPTION
**Pull request by StevenLocke**
Also limited the events being shown.

@Calligre/frontend

### CODE REVIEW

File: src/js/pages/attendee/Home.jsx [hunk 1]
```diff
@@ -149,19 +147,23 @@ export default class Featured extends React.Component {
...
-      if (moment().diff(moment(event.starttime), "hours") < 2 || eventCount < 3) {
+      const hourDiff = moment(event.starttime).diff(now, 'hours');
+      if ((hourDiff >= -1 && hourDiff <= 2) || eventCount < 3) {
+
         eventCount++;
-        return <Card type="event" key={"event-" + event.id} item={event}/>;
+        return <Card type="event" key={`event-${event.id}`} item={event} />;
       }
     });
+    EventComponents.sort((a, b) => a.props.item.starttime - b.props.item.starttime);
```
Comment (larajanecka) [inline src/js/pages/attendee/Home.jsx]: we might want to use `moment(a.props.item.starttime).isBefore` to stick with the moment stuff. Nothing wrong with this approach either.

**Review by larajanecka — APPROVED** [review]
Have I ever mentioned that I love you for linting things?

**Comment by StevenLocke** [comment]
@Calligre/frontend One more check please. I want to make sure sorting on events in state is ok. It feels like that's something we avoided but I don't remember why.

File: src/js/pages/attendee/Home.jsx (hunk 1 above)
Comment (StevenLocke) [reply src/js/pages/attendee/Home.jsx]: :+1: I like moment for date things. I swapped it over, though now it needs a conditional. Maybe better?

**Comment by larajanecka** [comment]
So long as you aren't calling setstate in the render function it should be fin. Lgtm

**Pull request merged**